Src/Newtonsoft.Json/JsonReader.cs

```diff
@@ -1109,5 +987,50 @@ public virtual void Close()
             _tokenType = JsonToken.None;
             _value = null;
         }
+
+        internal void ReadAndAssert()
+        {
+            if (!Read())
+            {
+                throw JsonSerializationException.Create(this, "Unexpected end.");
+            }
+        }
+
+        internal bool ReadAndSkipComments()
+        {
+            return Read() && SkipComments();
+        }
+
+        internal bool SkipComments()
```

Comment:
Considering that analogy with `XmlReader`, `MoveToContent` would be a closer name, with `ReadToContent` being more like what is `ReadAndSkipComments` here.

They could be useful public methods, though to completely match expected behaviour, it would need to start with `if (TokenType == JsonToken.None && !Read()) return false;`. That would be a bit of a waste on the internal uses, but perhaps still worth it.